PathOfBuildingCommunity/PathOfBuilding
# Add support for showing ineligible configurations

Fixes #5944 

### Description of the problem being solved:
Users may want to theory craft builds with configs that don't apply to their current setup and therefore aren't shown. This PR adds a toggle to the ConfigTab that allows an internal list of configs to be shown regardless of eligibility.

### Steps taken to verify a working solution:
- click toggle and verify configs in list are shown
- toggle again to make sure those configs are hidden
- add custom modifiers for other config tags like ifCond to make sure those still show/hide correctly
- save/reload builds with hidden configs checked to make sure they persist
- loaded 3.20 build

### Link to a build that showcases this PR:
Should work on any build, it's a UI change
<pre>eNqtW91z2jgQf-79FR7eG8Jnkg65GxJIwkzScJCmd08dYQvQRZYYS05C__pbSTY4BBkZuw-pQfvbL2lXu7Lo_fUeUu8VR4JwdllrnJzWPMx8HhC2uKz9eLr5el77688_emMkl4_zq5hQNfLnH196-tmj-BXTy1qz5kkULbB8Tjm1fgGnFWJyiTl7QP_x6JYHl7XvnOGaN0MsIDL95FMkxHcU4sva1AdwzUPCxyy43n6fEC5RhHyJo3sltR9L_sADGJ0jKmA4RIRNuf-C5W3E4xVYU_NeCX4zRKOH8ePkKaMTYVmdwKQvvTFFaxxNJZKegD-XtT54Bi3wAIXwF7ghGmPlpW73otPtNBtnrYtGq1bPBV_FkZDHcZiuMA4yoKaNcBzh4XyOfUle8XVE5PUSMX8r7tSGK0r7EFNJVpTgKKNVx4a4-8T8omujfeIS0cF4uiFtnpyettpn583u6Xn3EI7Lw_r_JHJ5RcGhR0hR2NGCEYmPBI85EZyVsC8LtU9RTCmEpxPtBAscvSJJPqpl583DGWFHee8BMXTNhcMcKcoxjiD6ZSHAFPscEkZRGQWR92SO3SkL2ZEAimpznB3DqStdYcbHKTSB3OhGOeUxdaSU2xzVtFIN8LsD1YhJJ16vXOr965BuOqiHd-MN5dnpyUWrc9E97Vw0Gi3rfjBergXxEX1A7ySMQ8ivT-gFbwWeNe2LZbGUDHKDDdq2ZvEbEuHiqGtOgyNQS8RFcZgKBQcnwHbqf1O0I-a7xdcPFuk0mdmF83w8xxNY9Wrbn1HsiNiKSGJnu7OenuaLWmCWyFu7mXOPsb-8hTJpgiR2y5Qbqk43162K1smtinCPWztdN0ABJyngfic18iAFnTRkOFqsp0uCaVCMOlXrGq0c0plychbt5OyP4gqtlyy0oEt-oihwS_pFdXpFIptgG_lrMiF3W5YYSkMABNi1Hh5H_D9VcdNisH4U8jhynHBD7GRAujeYTmOCg9h324yuKPRMrtqDVpQWQvSlRP7LgAcLXEhIccQ0Xq0gTagpd8WpvQ0KYZIpKL52HagfYb06ha3aBt0FbKmdBWw2dncpOxB3W9TmXMCYLbmziE0T-wAZIYQ8rztgaOAPp-4b6H2cGhlN6NhQjfkbaL5URxuiGDUUMRuAtZ67iTD7vXbm_4HcScCQBXGkQsFZxi5in5gnEkK2FGKAJPKCpOp9RhFBTDb1sYvAKPKX9zD1N4jSGSSAy1r2W_VpB9hI57ZX10dL6mkUrngk9ZfXiPpCSx-xVSw9ps-FQiL8X7N4PldnPCBARvrcanhzM7x-Gj0PE4WzEPFCKP3F4nCmDjHM_9tlNcU6a3oingnzeFl7JvhtqlADLBGh4ESfU4pWAgebsycCj5pmCp7xZQ43TQW1bXoitJ_XlsDOafiOIwn2wnbrRwRb9dqMH1DKCFRbsZpgGzd11mJnZLL9NRLSlAoWT-lzLTsXddRkNUcN5mBhzSBqlZyMHvCEXK-wgP2XzImvYiN_yp-A2lDl-MX3Iaz8dc58J3uWnYc-vLIxMIN2sDmAsqGT0Ryv6sMvq1fNqB0-wD6y2m4G7eBN9ckZuMnGZUOVw-k7Z3qRQ9D0CVX7i3VmhxRvSOwMH-USR2bTsnJ6gByVkuQGTkRmsbSHcYYix1e6E7Z4SI3Zoabbs9igxnIy0YceyOLQLI2dlekdrIksD2pqDav_ksolZwqSytzifjOa44S0ObHYnwznBInOv_1XTgJTvVrCZYcsL2FAnV6eja7Ey7PZrdHLc7yhSLxY5zsZtcN_QO9G5HoPFxnFrkxUUJXjoGKrHIfJbiGxxU7ySwidvwZ4jsHRuQlsQ5MzvzJmA9jkZM7cOrLSau0PhK1phXiZVL7X0sIczQJNzibz1rAhOcAI9qK7nGrHjdOmb7rDiKq3TZyWY_jpDLaUnVwKxIKBOuwpaag6K4pXwCzV7HFfubqd0l2uvXraSPR07SY8AR3GLQ7F1Ro6whu1O-2860LasUlxL017k8Wk4gI8RzFV3_8dI5VYVJeV-fbevCpnPAo3J_jAasnfVFI0HFUpKWCvub83I30qE2ZKXOov451EIW1m2jkZm9TjU4RxqjpUsorEwOBD9v36KNBqhlCx42id1CmqPWPQUcFD5_y81ap5Ethl3u03G8lrewNXnL_0fkzu9cOXpZQr8a1ef3t7O1khueRz_E4oPvF5WF8BCHT6qtuwr4ptvQ__rhZ9_U8zqqeceuZ1vjDroa5012Yq49TDdw7VkBpTX6YftOmqZ0v6zamMlD2_OQ__1baqp2QyWskEQPU-IGB7pGM_nVFF-E_aK_dGEqYgcal6ThdDLLB51_UToxVn-mtGqHF3QmglSmYvsZZymE5AjGfgAT0JumW9pdCUQeCoIePwej7ACPEaBSBXHILU68_WQiDqGb97nSIMMJW7-GYBvN7lvNYRVnrTN7TaFd0-hlEJ8xMezQp47DeoW4FB3QoMOorHPoNKr85WBR5pVmVNkfV2h6EakaX8sSfYuqU1aBbmUFpkqyr_dypYDI0KwqNRgR7tgiuhqhTTLBuR7QocWHzPqCzFtsra3ywZwsUUCNaeOZQoswZNbVGGw_7A7pTm0K1qWhtV5ZjiS7O8H9ul57JTzaIqYnw_jCmWFWTCVgUJpVUyKNsl8Y1q_H_0Tll40bYLI46pokp6tVM2V1c0Le2qkktldfXRLVQB3ISwRaEp3J-amtXMQWU7RRUNX_s4m4o6v_weX1Gx0SmtSPktrls0GZRNHp8F9urJUY8-oNJnRfraAmdzsvh0CWH3hzHbqwv6xyifby1EWIDvfaxecH6869B_Hk76t_tuOuz-0uWgEMxwuB6JKy6EG-2U_MZjdTNTbukfcEDicA_C_OhnL-MxRT5echrg6AN78_OX9I7GWfaq5z56dTMgvVaRgloHMGn0pfTN7mEZ6e2yFNPJhyQ38jilE8Syup0d0G3zRjEFnB9Q7uMdwAwuPftOVmOvvvt7rv8BsqA1Rg==</pre>

### After screenshot:
![image](https://user-images.githubusercontent.com/92683202/230444628-0508b298-1189-4db5-8a8b-23ff344d79af.png)
![image](https://user-images.githubusercontent.com/92683202/230444676-f460545b-aa4d-487a-b5e0-a9da73fc4d08.png)
![toggle2](https://user-images.githubusercontent.com/92683202/230444900-6d39dded-86ef-4bdb-8575-075f95346b71.gif)
